ios/FluentUI/Bottom Sheet/CommandingItem.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// An object representing a command.
+///
+/// `CommandingItem` defines the high level properties and behavior of a command. Its visual representation is determined by
+/// the `BottomCommandingController`.
+@objc(MSFCommandingItem)
+open class CommandingItem: NSObject {
+
+    /// A closure that's called when the command is triggered
+    @objc open var action: (CommandingItem) -> Void
+
+    /// The title of the command item.
+    @objc open var title: String {
+        didSet {
+            if title != oldValue {
+                delegate?.commandingItem(self, didChangeTitleFrom: oldValue)
+            }
+        }
+    }
+
+    /// A `UIImage` to be displayed with the command.
+    @objc open var image: UIImage {
+        didSet {
+            if image != oldValue {
+                delegate?.commandingItem(self, didChangeImageFrom: oldValue)
+            }
+        }
+    }
+
+    /// A `UIImage` used when the command is represented as a button in selected state.
+    @objc open var selectedImage: UIImage? {
+        didSet {
+            if selectedImage != oldValue {
+                delegate?.commandingItem(self, didChangeSelectedImageFrom: oldValue)
+            }
+        }
+    }
+
+    /// Indicates whether the command is currently on.
+    ///
+    /// When `commandType` is set to `.toggle`, the item toggles this automatically before calling `action`.
+    @objc open var isOn: Bool {
+        didSet {
+            if isOn != oldValue {
+                delegate?.commandingItem(self, didChangeOnFrom: oldValue)
+            }
+        }
+    }
+
+    /// Indicates whether the command is enabled.
+    @objc open var isEnabled: Bool {
+        didSet {
+            if isEnabled != oldValue {
+                delegate?.commandingItem(self, didChangeEnabledFrom: oldValue)
+            }
+        }
+    }
+
+    /// Determines the behavior of this command when its triggered.
+    @objc open var commandType: CommandType {
+        didSet {
+            if commandType != oldValue {
+                delegate?.commandingItem(self, didChangeCommandTypeFrom: oldValue)
+            }
+        }
+    }
+
+    @objc public init(title: String, image: UIImage, action: @escaping (CommandingItem) -> Void, selectedImage: UIImage? = nil, isSelected: Bool = false, isEnabled: Bool = true, commandType: CommandType = .simple) {
```

Comment:
What does @escaping do?

Reply:
It means that the closure can be called after the function returns. So any callback like this needs to be marked escaping, and Swift won't let you not mark it that way.